internal/pkg/service/buffer/storage/level/local/writer/base.go

```diff
@@ -19,16 +19,18 @@ type syncer = disksync.Syncer
 type BaseWriter struct {
 	*chain
 	*syncer
+	events   *Events
```

Comment:
Potrebujem nejak napojit `statistics` na `local.Writer`, hlavne riesim, aby sa po `Writer.Close` ulozila posledna hodnota statistik do databazy. To sa potom pouziva na vyhodnotenie `conditions`.

Nechcel som vytvarat nejake priame spojenie medzi `local/writer` a `statistics` pkg.

Preto som pridal `writer.Events`: `OnWriterOpen`/`OnWriterClose`.

V tomto PR je iba pridanie tychto eventov + testy.